pyproject.toml

```diff
@@ -71,6 +71,12 @@ exclude = ["ndcube._dev*"]
 [tool.setuptools_scm]
 write_to = "ndcube/_version.py"
 
+[project.entry-points."asdf.resource_mappings"]
+ndcube = "ndcube.asdf.entry_points:get_resource_mappings"
+
+[project.entry-points."asdf.extensions"]
+ndcube = "ndcube.asdf.entry_points:get_extensions"
+
```

Comment:
Would you also enable schema tests for the new schemas? Here's the configuration for `asdf-astropy`:
https://github.com/astropy/asdf-astropy/blob/1b286108042c1201199e9931cd2a5b52cc57bdce/pyproject.toml#L121-L122

I am not familiar with the ndcube tox and github actions CI but would you look into modifying whatever is necessary there to get the schema tests running for the PR? Let me know if there are more details or anything I can do to help with this. Also, perhaps one of the other maintainers has a sense for how these schema tests might be integrated in the CI.

Reply:
I have enabled the schema test, @Cadair could you please verify if the configuration is correct?
I followed [this](https://asdf.readthedocs.io/en/latest/asdf/extending/schemas.html#testing-custom-schemas:~:text=%5Btool.pytest.ini_options%5D%0Aasdf_schema_root%20%3D%20%27path/to/schemas%20another/path/to/schemas%27) to enable the tests

Reply:
I don't think this has allowed the schema tests to run in the CI. I think it's related to https://github.com/asdf-format/asdf/pull/1756

The core issue is that the CI run is testing files in the installed package but is running from a [temporary directory](https://github.com/sunpy/ndcube/blob/ce4d0b13705fcb8128a87402a23ac0378e7c58da/tox.ini#L19). Take [this test](https://github.com/sunpy/ndcube/actions/runs/9323888973/job/25668021958?pr=708#step:10:110) as an example:
```
../../.tox/py312/lib/python3.12/site-packages/ndcube/extra_coords/tests/test_extra_coords.py::test_empty_ec PASSED
```
The path to the file contains a bunch of stuff that is very specific to that test run (py312, tox, etc) and this causes the asdf pytest plugin to ignore the schema files since they appear at a similar path which doesn't match the `asdf_schema_root`.

As there are test configurations where we might expect the schema tests to fail (like the [oldest deps job](https://github.com/sunpy/ndcube/blob/ce4d0b13705fcb8128a87402a23ac0378e7c58da/tox.ini#L22), see [this conversation](https://github.com/DKISTDC/dkist/pull/324#issuecomment-2023135345) for more details) we may want to consider adding a new CI job (or jobs) that test just the schemas. Minimally they could be tested against the newest released asdf and perhaps they could also be tested against development versions of asdf and the dependent schema packages. Ignoring the development testing at the moment I think this would involve:
- add a tox factor that runs just the schema tests (defining `asdf_schema_root`, providing a path to the schemas, not running in a temporary directory)
- enabling this factor in the [CI workflow](https://github.com/sunpy/ndcube/blob/ce4d0b13705fcb8128a87402a23ac0378e7c58da/.github/workflows/ci.yml#L54)

Reply:
I've included a separate environment for schema testing in tox and created a workflow for it. Should I add a separate job for the schema tests as well? Please let me know. (P.S. I got a bit carried away with the workflow concept.)

Reply:
The workflow and tox changes you've included so far look good enough to me. I don't think adding a separate job is necessary (but please let me know if I'm missing something that would provide). Thanks!